run_project_tests.py

```diff
@@ -696,12 +700,20 @@ def _run_tests(all_tests, log_name_base, failfast: bool, extra_args):
             testname = '%.3d %s' % (int(testnum), testbase)
             should_fail = False
             suite_args = []
+            final_extra = extra_args.copy()
             if name.startswith('failing'):
                 should_fail = name.split('failing-')[1]
             if name.startswith('warning'):
                 suite_args = ['--fatal-meson-warnings']
                 should_fail = name.split('warning-')[1]
-            result = executor.submit(run_test, skipped, t.as_posix(), extra_args + suite_args,
+            if (name in {'cmake', 'common', 'platform-linux'} and
+                system_compiler in {'gcc', 'clang'} and
+                    t.name not in {'99 threads', '177 initial c_args'}):
+                final_extra += ['-Dc_args=-Werror=unused-parameter -Werror=return-type']
+            if name == 'fortran' and system_compiler == 'gcc':
+                final_extra += ['-Dfortran_args=-fimplicit-none']
```

Comment:
Thinking about this further, I'm not sure this is what we should be doing. The basic test suite should be run without any external test arguments _because_ that is how end users will use their projects. It is entirely possible to write a bug that depends on these arguments always being in `c_args`, because those arguments are always use in e.g. compiler checks.

Reply:
Yes so perhaps a solution that addresses both scenarios is to:

1. remove those internal -Dc_args and -Dfortran_args arguments that were initially added here
2. set a single Azure pipeline that just runs those unused parameters  from the command line

Reply:
Seems reasonable. Simply setting `CFLAGS` et al envvars for one run should do it.